In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import os
os.chdir("/content/drive/My Drive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC")


In [ ]:
!pwd

/content/drive/My Drive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC


In [ ]:
import os
os.environ['PATH'] += ":/content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin"

In [ ]:
!chmod +x /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx

In [ ]:
!gmx pdb2gmx -f 5BRZ_cleaned.pdb -o 5BRZ_processed.gro -water tip3p

                     :-) GROMACS - gmx pdb2gmx, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx pdb2gmx -f 5BRZ_cleaned.pdb -o 5BRZ_processed.gro -water tip3p

Select the Force Field:

From '/content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/share/top':

 1: AMBER03 protein, nucleic AMBER94 (Duan et al., J. Comp. Chem. 24, 1999-2012, 2003)

 2: AMBER94 force field (Cornell et al., JACS 117, 5179-5197, 1995)

 3: AMBER96 protein, nucleic AMBER94 (Kollman et al., Acc. Chem. Res. 29, 461-469, 1996)

 4: AMBER99 protein, nucleic AMBER94 (Wang et al., J. Comp. Chem. 21, 1049-1074, 2000)

 5: AMBER99SB protein, nucleic AMBER94 (Hornak et al., Proteins 65, 712-725, 2006)

 6: AMBER99SB-ILDN protein, nucleic AMBER94 (Lindorff-Larsen et al., Prot

In [ ]:
!gmx pdb2gmx -f 5BRZ_cleaned.pdb -o 5BRZ_processed.gro -water tip3p -ff amber99sb-ildn

                     :-) GROMACS - gmx pdb2gmx, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx pdb2gmx -f 5BRZ_cleaned.pdb -o 5BRZ_processed.gro -water tip3p -ff amber99sb-ildn

Using the Amber99sb-ildn force field in directory amber99sb-ildn.ff

going to rename amber99sb-ildn.ff/aminoacids.r2b
Opening force field file /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/share/top/amber99sb-ildn.ff/aminoacids.r2b

going to rename amber99sb-ildn.ff/dna.r2b
Opening force field file /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/share/top/amber99sb-ildn.ff/dna.r2b

going to rename amber99sb-ildn.ff/rna.r2b
Opening force field file /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/share/top/amber99sb-ildn.ff/rna.r2b
Reading 5BRZ_

In [ ]:
!gmx editconf -f 5BRZ_processed.gro -o 5BRZ_newbox.gro -c -d 16 -bt cubic

                     :-) GROMACS - gmx editconf, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx editconf -f 5BRZ_processed.gro -o 5BRZ_newbox.gro -c -d 16 -bt cubic

Note that major changes are planned in future for editconf, to improve usability and utility.
Read 12808 atoms
Volume: 863.132 nm^3, corresponds to roughly 388400 electrons
No velocities found
    system size :  8.154  8.267 12.805 (nm)
    diameter    : 14.082               (nm)
    center      : 13.156  2.833 15.273 (nm)
    box vectors :  8.154  8.267 12.805 (nm)
    box angles  :  90.00  90.00  90.00 (degrees)
    box volume  : 863.13               (nm^3)
    shift       :  9.885 20.208  7.768 (nm)
new center      : 23.041 23.041 23.041 (nm)
new box vectors : 46.082 46.082 

In [ ]:
!gmx editconf -f 5BRZ_processed.gro -o 5BRZ_newbox.gro -c -d 1.6 -bt cubic

                     :-) GROMACS - gmx editconf, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx editconf -f 5BRZ_processed.gro -o 5BRZ_newbox.gro -c -d 1.6 -bt cubic

Note that major changes are planned in future for editconf, to improve usability and utility.
Read 12808 atoms
Volume: 863.132 nm^3, corresponds to roughly 388400 electrons
No velocities found
    system size :  8.154  8.267 12.805 (nm)
    diameter    : 14.082               (nm)
    center      : 13.156  2.833 15.273 (nm)
    box vectors :  8.154  8.267 12.805 (nm)
    box angles  :  90.00  90.00  90.00 (degrees)
    box volume  : 863.13               (nm^3)
    shift       : -4.515  5.808 -6.632 (nm)
new center      :  8.641  8.641  8.641 (nm)
new box vectors : 17.282 17.282

In [ ]:
!gmx solvate -cp 5BRZ_newbox.gro -cs spc216.gro -o 5BRZ_solvated.gro -p topol.top

                     :-) GROMACS - gmx solvate, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx solvate -cp 5BRZ_newbox.gro -cs spc216.gro -o 5BRZ_solvated.gro -p topol.top

Reading solute configuration
Reading solvent configuration

Initialising inter-atomic distances...

         based on residue and atom names, since they could not be
         definitively assigned from the information in your input
         files. These guessed numbers might deviate from the mass
         and radius of the atom type. Please check the output
         files if necessary. Note, that this functionality may
         be removed in a future GROMACS version. Please, consider
         using another file format for your input.

NOTE: From version 5.0 gmx solvate u

In [ ]:
mdp_content = """
; ions.mdp - used for adding ions to neutralize the system
integrator  = steep     ; Algorithm for energy minimization.
emtol       = 1000.0    ; Minimization stops when the maximum force < 1000.0 kJ/mol/nm
emstep      = 0.01      ; Energy minimization step size.
nsteps      = 50000     ; Maximum number of (minimization) steps.
nstlist     = 1         ; Frequency to update the neighbor list, every 1 steps
ns_type     = grid      ; Method to determine neighbor list (simple grid)
rlist       = 1.0       ; Cut-off for making neighbor list (short-range interactions)
coulombtype = cutoff    ; Simple cut-off scheme for Coulomb interactions
rcoulomb    = 1.0       ; Short-range electrostatic cut-off
rvdw        = 1.0       ; Short-range van der Waals cut-off
pbc         = xyz       ; Periodic boundary conditions in all directions
"""

with open('ions.mdp', 'w') as mdp_file:
    mdp_file.write(mdp_content)


In [ ]:
!gmx grompp -f ions.mdp -c 5BRZ_solvated.gro -p topol.top -o ions.tpr

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f ions.mdp -c 5BRZ_solvated.gro -p topol.top -o ions.tpr

Ignoring obsolete mdp entry 'ns_type'

NOTE 1 [file ions.mdp]:
  With Verlet lists the optimal nstlist is >= 10, with GPUs >= 20. Note
  that with the Verlet scheme, nstlist has no effect on the accuracy of
  your simulation.

Setting the LD random seed to 1637724030

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

Excluding 3 bonded neighbours molecule type 'Protein_chain_B'

Excluding 3 bonded neighbours molecu

In [ ]:
!echo "integrator = steep" > ions.mdp
!echo "emtol = 1000.0" >> ions.mdp
!echo "emstep = 0.01" >> ions.mdp
!echo "nsteps = 50000" >> ions.mdp
!echo "nstlist = 20" >> ions.mdp
!echo "rlist = 1.0" >> ions.mdp
!echo "coulombtype = PME" >> ions.mdp
!echo "rcoulomb = 1.0" >> ions.mdp
!echo "rvdw = 1.0" >> ions.mdp
!echo "pbc = xyz" >> ions.mdp

In [ ]:
!gmx grompp -f ions.mdp -c 5BRZ_solvated.gro -p topol.top -o ions.tpr

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f ions.mdp -c 5BRZ_solvated.gro -p topol.top -o ions.tpr

Setting the LD random seed to -1703943

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

Excluding 3 bonded neighbours molecule type 'Protein_chain_B'

Excluding 3 bonded neighbours molecule type 'Protein_chain_C'

Excluding 3 bonded neighbours molecule type 'Protein_chain_D'

Excluding 3 bonded neighbours molecule type 'Protein_chain_E'

Excluding 2 bonded neighbours molecule type 'SOL'

NOTE 1 [file topol.top, l

In [ ]:
!echo "SOL" | gmx genion -s ions.tpr -o 5BRZ_solv_ions.gro -p topol.top -pname NA -nname CL -np 15 -neutral

                      :-) GROMACS - gmx genion, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx genion -s ions.tpr -o 5BRZ_solv_ions.gro -p topol.top -pname NA -nname CL -np 15 -neutral

Reading file ions.tpr, VERSION 2023.2 (single precision)
Reading file ions.tpr, VERSION 2023.2 (single precision)
Will try to add 15 NA ions and 0 CL ions.
Select a continuous group of solvent molecules
Group     0 (         System) has 510403 elements
Group     1 (        Protein) has 12808 elements
Group     2 (      Protein-H) has  6567 elements
Group     3 (        C-alpha) has   822 elements
Group     4 (       Backbone) has  2466 elements
Group     5 (      MainChain) has  3293 elements
Group     6 (   MainChain+Cb) has  4061 elements
Group     7 (    

In [ ]:
!gmx grompp -f ions.mdp -c 5BRZ_solv_ions.gro -p topol.top -o ions.tpr -maxwarn 1

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f ions.mdp -c 5BRZ_solv_ions.gro -p topol.top -o ions.tpr -maxwarn 1

Setting the LD random seed to -550184021

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

Excluding 3 bonded neighbours molecule type 'Protein_chain_B'

Excluding 3 bonded neighbours molecule type 'Protein_chain_C'

Excluding 3 bonded neighbours molecule type 'Protein_chain_D'

Excluding 3 bonded neighbours molecule type 'Protein_chain_E'

Excluding 2 bonded neighbours molecule type 'SOL'

Excluding 1

In [ ]:
!gmx grompp -f minim.mdp -c 5BRZ_solv_ions.gro -p topol.top -o em.tpr

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f minim.mdp -c 5BRZ_solv_ions.gro -p topol.top -o em.tpr

Ignoring obsolete mdp entry 'ns_type'

NOTE 1 [file minim.mdp]:
  With Verlet lists the optimal nstlist is >= 10, with GPUs >= 20. Note
  that with the Verlet scheme, nstlist has no effect on the accuracy of
  your simulation.

Setting the LD random seed to -270672205

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

turning all bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_

In [ ]:
!gmx grompp -f minim.mdp -c 5BRZ_solv_ions.gro -p topol.top -o em.tpr

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f minim.mdp -c 5BRZ_solv_ions.gro -p topol.top -o em.tpr

Setting the LD random seed to -41980169

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_B'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_C'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_D'

turning H bonds into constr

In [ ]:
!gmx mdrun -v -deffnm em

                      :-) GROMACS - gmx mdrun, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx mdrun -v -deffnm em

Compiled SIMD is AVX2_256, but CPU also supports AVX_512 (see log).
Reading file em.tpr, VERSION 2023.2 (single precision)
Using 1 MPI thread
Using 12 OpenMP threads 


Steepest Descents:
   Tolerance (Fmax)   =  1.00000e+03
   Number of steps    =        50000
Step=    0, Dmax= 1.0e-02 nm, Epot= -5.47057e+06 Fmax= 3.38375e+05, atom= 8554
Step=    1, Dmax= 1.0e-02 nm, Epot= -5.60532e+06 Fmax= 1.03728e+05, atom= 253835
Step=    2, Dmax= 1.2e-02 nm, Epot= -5.84826e+06 Fmax= 4.13767e+04, atom= 259457
Step=    3, Dmax= 1.4e-02 nm, Epot= -6.08950e+06 Fmax= 1.83718e+04, atom= 147761
Step=    4, Dmax= 1.7e-02 nm, Epot= -6.30987e+06 Fm

The output indicates that your energy minimization process has successfully converged, which is great news. Here’s a breakdown of what the results mean and what your next steps should be:

Analysis of Energy Minimization Output
Step and Dmax: These values provide information about each step of the energy minimization. Dmax is the maximum step size in nm (nanometers) that the algorithm uses for updating atomic positions to minimize the potential energy.
Epot (Potential Energy): The potential energy of the system is decreasing, indicating that the system is moving towards a more stable, low-energy configuration.
Fmax (Maximum Force): The maximum force acting on any atom decreases progressively, which is the criterion for the convergence of the minimization process. The final reported maximum force is 955.0954, which is less than the convergence criterion (Fmax < 1000), indicating successful minimization.
Atom: The specific atom (1844) consistently experiences the highest force during these steps, which might be a point of interest if you were troubleshooting issues with the simulation or interested in specific interactions involving this atom.



In [ ]:
!gmx grompp -f nvt.mdp -c em.gro -p topol.top -o nvt.tpr

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f nvt.mdp -c em.gro -p topol.top -o nvt.tpr

Ignoring obsolete mdp entry 'ns_type'
Replacing old mdp entry 'nstxtcout' by 'nstxout-compressed'
Setting the LD random seed to 1978531261

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

turning all bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_B'

turning all bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_C'

turning all bonds into constraints...

In [ ]:
!gmx mdrun -v -deffnm nvt

                      :-) GROMACS - gmx mdrun, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx mdrun -v -deffnm nvt

Compiled SIMD is AVX2_256, but CPU also supports AVX_512 (see log).
Reading file nvt.tpr, VERSION 2023.2 (single precision)
Changing nstlist from 10 to 80, rlist from 1.2 to 1.321

Update groups can not be used for this system because there are three or more consecutively coupled constraints

Using 1 MPI thread
Using 12 OpenMP threads 

starting mdrun 'Protein in water'
50000 steps,    100.0 ps.
step 500, will finish Sun May  5 19:05:17 2024

Received the second INT/TERM signal, stopping within 11 steps

step 531, will finish Sun May  5 19:04:27 2024
               Core t (s)   Wall t (s)        (%)
       Time:      880.172  

In [ ]:
!gmx grompp -f nvt.mdp -c em.gro -p topol.top -o nvt.tpr

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f nvt.mdp -c em.gro -p topol.top -o nvt.tpr

Ignoring obsolete mdp entry 'ns_type'
Setting the LD random seed to -1074266241

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_B'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_C'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_D'



In [ ]:
!gmx mdrun -v -deffnm nvt

                      :-) GROMACS - gmx mdrun, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx mdrun -v -deffnm nvt


Back Off! I just backed up nvt.log to ./#nvt.log.1#
Compiled SIMD is AVX2_256, but CPU also supports AVX_512 (see log).
Reading file nvt.tpr, VERSION 2023.2 (single precision)
Changing nstlist from 10 to 80, rlist from 1.2 to 1.321

Using 1 MPI thread
Using 12 OpenMP threads 


Back Off! I just backed up nvt.xtc to ./#nvt.xtc.1#

Back Off! I just backed up nvt.edr to ./#nvt.edr.1#
starting mdrun 'Protein in water'
10000 steps,     20.0 ps.
step 9900, remaining wall clock time:    13 s          
Writing final coordinates.
step 10000, remaining wall clock time:     0 s          
               Core t (s)   Wall t (s)        (%)

Breakdown of the Output:
Performance Metrics: The simulation achieved a performance of approximately 1.282 ns/day, which is reasonable given the resources mentioned (using one MPI thread and 12 OpenMP threads on a possibly non-specialized compute environment like Google Colab).
Backup Notifications: The logs (nvt.log), trajectory (nvt.xtc), and energy data (nvt.edr) files have been backed up, indicating that new outputs were generated and saved, replacing any older files.
Simulation Details: The message about changing nstlist from 10 to 80 and adjusting rlist reflects GROMACS’ internal optimizations to enhance performance based on your system’s setup and the computing environment.

In [ ]:
!echo "help" | gmx energy -f nvt.edr > energy_terms.txt

                      :-) GROMACS - gmx energy, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx energy -f nvt.edr

Opened nvt.edr as single precision energy file

Select the terms you want from the following list by
selecting either (part of) the name or the number or a combination.
End your selection with an empty line or a zero.
-------------------------------------------------------------------
  1  Bond             2  Angle            3  Proper-Dih.      4  Per.-Imp.-Dih.
  5  LJ-14            6  Coulomb-14       7  LJ-(SR)          8  Coulomb-(SR)  
  9  Coul.-recip.    10  Potential       11  Kinetic-En.     12  Total-Energy  
 13  Conserved-En.   14  Temperature     15  Pressure        16  Constr.-rmsd  
 17  Vir-XX          18  Vir-X

In [ ]:
!cat energy_terms.txt

In [ ]:
!gmx grompp -f npt.mdp -c nvt.gro -t nvt.cpt -p topol.top -o npt.tpr

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f npt.mdp -c nvt.gro -t nvt.cpt -p topol.top -o npt.tpr

Ignoring obsolete mdp entry 'ns_type'
Setting the LD random seed to -1096812033

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_B'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_C'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protei

In [ ]:
!gmx mdrun -v -deffnm npt

                      :-) GROMACS - gmx mdrun, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx mdrun -v -deffnm npt

Compiled SIMD is AVX2_256, but CPU also supports AVX_512 (see log).
Reading file npt.tpr, VERSION 2023.2 (single precision)
Changing nstlist from 10 to 80, rlist from 1.2 to 1.321

Using 1 MPI thread
Using 12 OpenMP threads 

starting mdrun 'Protein in water'
10000 steps,     20.0 ps.
step 9900, remaining wall clock time:    13 s          
Writing final coordinates.
step 10000, remaining wall clock time:     0 s          
               Core t (s)   Wall t (s)        (%)
       Time:    16307.043     1358.938     1200.0
                 (ns/day)    (hour/ns)
Performance:        1.272       18.872

GROMACS reminds you: "I hadn'

Analysis of NPT Equilibration Output:
Performance Metrics: The simulation maintained a performance rate of 1.272 ns/day, which is consistent with the computational resources described (1 MPI thread and 12 OpenMP threads). This performance level is appropriate given the non-specialized compute environment.
System Stability: With the final coordinates written and no errors or stability issues noted in your output, it seems the system has reached a satisfactory state of equilibration over the simulated time.
Resource Management: The efficient use of the available CPU capabilities (AVX2_256, although AVX_512 is available but not utilized due to compatibility or configuration) shows optimal resource usage within the confines of your setup.

In [ ]:
!gmx grompp -f md.mdp -c npt.gro -t npt.cpt -p topol.top -o md_100ns.tpr

                      :-) GROMACS - gmx grompp, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx grompp -f md.mdp -c npt.gro -t npt.cpt -p topol.top -o md_100ns.tpr

Ignoring obsolete mdp entry 'ns_type'
Setting the LD random seed to -538968321

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_B'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Protein_chain_C'

turning H bonds into constraints...

Excluding 3 bonded neighbours molecule type 'Pro

In [ ]:
!gmx mdrun -v -deffnm md_100ns

                      :-) GROMACS - gmx mdrun, 2023.2 (-:

Executable:   /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2/build/bin/gmx
Data prefix:  /content/drive/MyDrive/gromacs-on-colab/gromacs-2023.2 (source tree)
Working dir:  /content/drive/MyDrive/gromacs-on-colab/Daily-notebook- GROMACS-Colab/5-5-24-TCR-MHC
Command line:
  gmx mdrun -v -deffnm md_100ns


Back Off! I just backed up md_100ns.log to ./#md_100ns.log.1#
Compiled SIMD is AVX2_256, but CPU also supports AVX_512 (see log).
Reading file md_100ns.tpr, VERSION 2023.2 (single precision)
Changing nstlist from 10 to 80, rlist from 1.2 to 1.321

Using 1 MPI thread
Using 12 OpenMP threads 


Back Off! I just backed up md_100ns.xtc to ./#md_100ns.xtc.1#

Back Off! I just backed up md_100ns.edr to ./#md_100ns.edr.1#


starting mdrun 'Protein in water'
1000000 steps,   2000.0 ps.
step 105900, will finish Tue May  7 07:58:33 2024

Received the second INT/TERM signal, stopping within 11 steps

step 105981, will finish Tue Ma